src/cascadia/WpfTerminalControl/TerminalContainer.cs

```diff
@@ -196,20 +193,42 @@ internal string GetSelectedText()
 
             NativeMethods.TerminalTriggerResizeWithDimension(this.terminal, dimensions, out dimensionsInPixels);
 
-            // If AutoFill is true, keep Rows and Columns in sync with MaxRows and MaxColumns.
-            // Otherwise, MaxRows and MaxColumns will be set on startup and on control resize by the user.
-            if (this.AutoFill)
-            {
-                this.MaxColumns = dimensions.X;
-                this.MaxRows = dimensions.Y;
-            }
-
             this.Columns = dimensions.X;
             this.Rows = dimensions.Y;
 
+            this.TerminalRendererSize = new Size()
+            {
+                Width = dimensionsInPixels.cx,
+                Height = dimensionsInPixels.cy,
+            };
+
             this.Connection?.Resize((uint)dimensions.Y, (uint)dimensions.X);
+        }
+
+        /// <summary>
+        /// Calculates the rows and columns that would fit in the given size.
+        /// </summary>
+        /// <param name="size">DPI scaled size.</param>
+        /// <returns>Amount of rows and columns that would fit the given size.</returns>
+        internal (uint columns, uint rows) CalculateRowsAndColumns(Size size)
+        {
+            NativeMethods.COORD dimensions;
+            NativeMethods.TerminalCalculateResize(this.terminal, (short)size.Width, (short)size.Height, out dimensions);

```

Comment:
nit: NativeMethods.COORD dimensions can be defined inline instead of on a new line.